optaplanner-docs/src/main/asciidoc/Development/Development-chapter.adoc

```diff
@@ -65,4 +69,25 @@ Maybe the member (getFooRange) should return CountableValueRange.
 +
 The word _maybe_ is to indicate that the advice is not guaranteed to be right in all cases.
 
-. Generics. The `Solution` class is often passed as a generic type parameter to subsystems. The `PlanningEntity` class(es) are rarely passed as a generic type parameter.
+=== Generics
+
+. The `Solution` class is often passed as a generic type parameter to subsystems.
+. The `PlanningEntity` class(es) are rarely passed as a generic type parameter because there could be multiple planning entities.
+
+=== Lifecycle
+
+One of the biggest challenges in multi-algorithm implementations (such as Planner)
```

Comment:
Should have been "OptaPlanner" ?

Reply:
All docs use "Planner" due to old product docs writers not wanting to see the word "OptaPlanner" when the product version was still called "Business Resource Planner". For consistency, I've continued to use that here, but ideally we should replace all "Planner" with "OptaPlanner" - or by a conditional that resolves to OptaPlanner in community and "Business Optimizer" in product.
In any case, different issue - so for now "keep it consistent" means doing it like this.